test_eos-phone-home.py

```diff
@@ -0,0 +1,159 @@
+import imp
+import os
+
+import pytest
+
+HERE = os.path.dirname(__file__)
+eos_phone_home_path = os.path.join(HERE, 'eos-phone-home')
+eos_phone_home = imp.load_source("m", eos_phone_home_path)
+
+
+@pytest.fixture
+def phone_home(tmpdir):
+    return eos_phone_home.PhoneHome(True, False, root=tmpdir.strpath)
+
+
+def test_instantiate(tmpdir, phone_home):
+    assert tmpdir.strpath == phone_home._root
+
+
+def test_get_product_info_arm(tmpdir, phone_home):
+    compatible = tmpdir.join('proc', 'device-tree', 'compatible')
+    compatible.write_binary(b'vendor1,model1\0vendor2,model2\0',
+                            ensure=True)
+
+    ssn = tmpdir.join('sys', 'class', 'endless_mfgdata', 'entries', 'SSN')
+    ssn.write('  serial  ', ensure=True)
+
+    pi = phone_home._get_product_info()
+    assert pi == eos_phone_home.ProductInfo('vendor1', 'model1', 'serial')
+
+
+def test_get_product_info_arm_missing_serial(tmpdir, phone_home):
+    compatible = tmpdir.join('proc', 'device-tree', 'compatible')
+    compatible.write_binary(b'vendor1,model1\0vendor2,model2\0',
+                            ensure=True)
+
+    pi = phone_home._get_product_info()
+    assert pi == eos_phone_home.ProductInfo('vendor1', 'model1', None)
+
+
+def test_get_product_info_arm_missing_dt(tmpdir, phone_home):
+    ssn = tmpdir.join('sys', 'class', 'endless_mfgdata', 'entries', 'SSN')
+    ssn.write('  serial  ', ensure=True)
+
+    pi = phone_home._get_product_info()
+    assert pi == eos_phone_home.ProductInfo(None, None, 'serial')
+
+
+def test_get_product_info_x86(tmpdir, phone_home):
+    scdi = tmpdir.join('sys', 'class', 'dmi', 'id')
+    scdi.join('sys_vendor').write('  vendor', ensure=True)
+    scdi.join('product_name').write(' product')
+    scdi.join('product_serial').write('serial  ')
+
+    pi = phone_home._get_product_info()
+    assert pi == eos_phone_home.ProductInfo('vendor', 'product', 'serial')
+
+
+def test_get_product_info_x86_missing_vendor(tmpdir, phone_home):
+    scdi = tmpdir.join('sys', 'class', 'dmi', 'id')
+    scdi.join('product_name').write(' product', ensure=True)
+    scdi.join('product_serial').write('serial  ')
+
+    pi = phone_home._get_product_info()
+    assert pi == eos_phone_home.ProductInfo(None, 'product', 'serial')
+
+
+def test_singleboot(tmpdir, phone_home):
```

Comment:
Nit: Maybe add what about `singleboot` this test is testing? Some of the other tests could also use this change.

Reply:
I have renamed these to test_{singleboot,dualboot,live}_system_is_detected